### ISSUE TITLE
$asyncValidators rejection message

### ISSUE THREAD

**Issue by klmdb** [angular/angular.js#13201]
Is there any way to display the `$asyncValidators` rejection message in the template?

I guess I could setup an isolated scope in my validation directive and then set the message to a bound variable, but that seems like a kind of hack outside the expected validator flow. 

The same case could be made for a validator that receives unexpected input. It seems like a validator should have 3 possible states instead of `{true, false}`: `{true, false, undefined}` with the undefined state exposing the error message somehow.

**Comment by Narretz** [comment]
Yes, it would make more sense if Validators could return metadata in errors / successes. Defintiely something to consider.
